recipe/meta.yaml

```diff
@@ -0,0 +1,161 @@
+    - m2-patch  # [win]
+    - patchelf  # [linux]
+    - protobuf {{ libprotobuf }}
+    - llvm-openmp  # [osx]
+    - intel-openmp  # [not osx and blas_impl == 'mkl']
+  host:
+    - python
+    - setuptools
+    - wheel
+    - pip
+    - pytorch 2.1.0
+    - pybind11
+    - libabseil
+    - ffmpeg 6.1.1
+  run:
+    - python
+    # according to compatibility matrix in
+    # https://github.com/pytorch/audio/blob/v2.1.2/docs/source/installation.rst#compatibility-matrix
+    - pytorch >=2.1.0,<2.2.0
+    # according to optional dependencies
+    # https://github.com/pytorch/audio/blob/v2.1.2/docs/source/installation.rst#optional-dependencies
+    # in this special case we want users of channel sfe140ed to be able to use the load and save functions, therefore
+    # at least ffmpeg (the mightiest of the backends) is specified as a run dependency.
+    - ffmpeg >=4.4,<7
+    # this is not officially required but it is suggested in some upstream build examples and allows the correct library to be used 
+    - libprotobuf <4.21.0
+  run_constrained:
+    # according to optional dependencies
+    # https://github.com/pytorch/audio/blob/main/docs/source/installation.rst#optional-dependencies
+    # sox has been tested with version 14.4.2, but it is unlikely that other versions will work
+    - sox >=14.4.2
+
+
+{% set ignore_modules =                  " --ignore=test/torchaudio_unittest/backend/soundfile --ignore=test/torchaudio_unittest/backend/dispatcher/soundfile" %}
+{% set ignore_modules = ignore_modules + " --ignore=test/torchaudio_unittest/backend/dispatcher/sox --ignore=test/torchaudio_unittest/backend/sox_io" %}
+
+# tests to ignore because of unsupported torchaudio function
+{% set tests_to_ignore = "test_rnnt_loss or test_rnnt_loss_0 or test_rnnt_loss_1 or test_rnnt_loss_2 or test_rnnt_loss_basic_backward or test_rnnt_loss_basic_forward_no_grad" %}
+{% set tests_to_ignore = tests_to_ignore + " or test_rnnt_loss_costs_and_gradients_0 or test_rnnt_loss_costs_and_gradients_1 or test_rnnt_loss_costs_and_gradients_2 or test_rnnt_loss_costs_and_gradients_3" %}
+{% set tests_to_ignore = tests_to_ignore + " or test_rnnt_loss_costs_and_gradients_random_data_with_numpy_fp32_0 or test_rnnt_loss_costs_and_gradients_random_data_with_numpy_fp32_1" %}
+{% set tests_to_ignore = tests_to_ignore + " or test_rnnt_loss_nonfused_softmax or test_rnnt_loss_basic_backward" %}
+# tests to ignore because of unexpected argument to load() function
+{% set tests_to_ignore = tests_to_ignore + " or test_tedlium_release1_path or test_tedlium_release1_str or test_tedlium_release2 or test_tedlium_release3" %}
+
+test:
+  source_files:
+    - test/torchaudio_unittest
+    - examples
+  imports:
+    - torchaudio
+  commands:
+    - pip check
+    # skipping osx x86_64 tests because of segfaults
+    - pip install expecttest dataset kaldi-io  # [not (osx and x86_64)]
+    - export PYTHONPATH=$PWD/examples/hubert:$PYTHONPATH  # [not (osx and x86_64)]
```

Comment:
is there a better way than using PWD? feels unsafe. Also `export` doesn't work on win

Reply:
true, I would have to use win syntax when this finally builds on `win-64`.
I will try to find some way around `$PWD`, but anyway this is just the test section.

Reply:
actually, as it turns out, `$PWD` is set by `conda-build` and is the only variable pointing to the `test_tmp` directory.